main.nf

```diff
@@ -33,23 +33,241 @@ ANSI_RESET = "\033[0m"
 params.input = false
+params.outdir = "./results"
+params.csv = false
 
 // Options: guppy basecalling
-params.input_path = false
+params.skip_basecalling = false
+params.skip_demultiplexing = false
+params.skip_porechop = false
+
+//params.input_path = false
 params.flowcell = false
 params.kit = false
-params.barcode_kit = false
-params.guppy_config = false
-params.guppy_model = false
+params.barcode_kits = false
+//params.guppy_config = false
+//params.guppy_model = false
+
+params.cpus = false
+params.cpu_threads_per_caller = false
+params.num_callers = false
+params.config = false
+params.trim_barcodes = false
+
+ch_input_files = Channel.fromPath( params.input )
+ch_input_csv = params.csv ? Channel.fromPath( params.csv, checkIfExists: true ) : Channel.empty()
 
+/*if ( params.csv ) { 
+  ch_input_csv = file(params.csv, checkIfExists: true) 
+} else { 
+  exit 1, "Samplesheet file not specified!" 
+}
+*/
+//ch_input_csv = Channel.fromPath( params.csv )
 //options: qc
 
 
 /*
 guppy basecalling
 */
+if ( !params.skip_basecalling ) {
+  
+  process guppy_basecaller {
+    publishDir path: params.barcode_kits ? "${params.outdir}/barcodes" : "${params.outdir}/basecalled", mode:'copy'
+    
+    input:
+    file dir_fast5 from ch_input_files
+    file csv_file from ch_input_csv.ifEmpty([])
+
+    output:
+    file "fastq/*.fastq.gz" into ch_fastq, ch_for_seqkit
+    file "guppy_basecaller.log" into ch_log_guppy_basecaller
+    file "rename.log" into ch_log_rename
+    file "results-guppy-basecaller/*.txt" into ch_summary_guppy
+
+    script:
+    flowcell = params.flowcell ? "--flowcell $params.flowcell" : ""
+    kit = params.kit ? "--kits $params.kit" : ""
+    barcode_kits = params.barcode_kits ? "--barcode_kits $params.barcode_kits" : ""
+    config = params.config ? "--config $params.config" : ""
+    trim_barcodes = params.trim_barcodes ? "--trim_barcodes" : ""
+
+    cpu_threads_per_caller = params.cpu_threads_per_caller ? "--cpu_threads_per_caller $params.cpu_threads_per_caller" 
+                          : (params.cpus ?  "--cpu_threads_per_caller $params.cpus" : "--cpu_threads_per_caller 2")
+    num_callers = params.num_callers ? "--num_callers $params.num_callers" : "--num_callers 1"
+
+    """
+    guppy_basecaller \\
+      --input_path $dir_fast5 \\
+      --save_path ./results-guppy-basecaller \\
+      --recursive \\
+      --records_per_fastq 0 \\
+      $flowcell \\
+      $kit \\
+      $barcode_kits \\
+      $trim_barcodes \\
+      $cpu_threads_per_caller \\
+      $num_callers \\
+      --qscore_filtering \\
+      $config \\
+      --compress_fastq \\
+      &> guppy_basecaller.log 
+
+    mkdir fastq
+    cd results-guppy-basecaller/pass
+    if [ "\$(find . -type d -name "barcode*" )" != "" ]
+    then
+      for dir in barcode*/
+      do
+        dir=\${dir%*/}
+        cat \$dir/*.fastq.gz > ../../fastq/\$dir.fastq.gz
+      done
+    else
+      cat *.fastq.gz > ../../fastq/unclassified.fastq.gz
```

Comment:
will have to check, but pretty sure `cat` on gz files is not going to work. Here it is always one fastq.gz file, so cat behaves like `mv`